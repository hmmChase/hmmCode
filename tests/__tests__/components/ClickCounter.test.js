import { render, prettyDOM, fireEvent } from '@testing-library/react';
import ClickCounter from '../../components/ClickCounter';

describe('ClickCounter', () => {
  it('matches snapshot', () => {
    const mockProps = {};

    const utils = render(<ClickCounter {...mockProps} />);

    const container = utils.container;

    expect(container).toMatchSnapshot();
  });

  it('increments number', () => {
    const mockProps = {};

    const utils = render(<ClickCounter {...mockProps} />);

    const button1 = utils.getByText('Click me');

    const span1 = utils.getByText('You clicked 0 times');

    expect(button1).toBeInTheDocument();
    expect(span1).toBeInTheDocument();

    fireEvent.click(button1);

    const span2 = utils.getByText('You clicked 1 times');

    expect(span2).toBeInTheDocument();
  });
});
