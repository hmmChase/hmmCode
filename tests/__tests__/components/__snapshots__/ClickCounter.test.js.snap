// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClickCounter matches snapshot 1`] = `
<div>
  <button>
    Click me
  </button>
  <span>
    You clicked 
    0
     times
  </span>
</div>
`;
